packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -53,13 +121,38 @@ export class State {
   public _onEnterEventsHaveAtLeastOneCondition(): boolean {
     return this.props.onEnter?.some(event => event.condition) ?? false;
   }
+
+  private toStateJson(): CfnDetectorModel.StateProperty {
+    const { stateName, onEnter } = this.props;
+    return {
+      stateName,
+      onInput: {
+        transitionEvents: toTransitionEventJson(this.transitionEvents),
+      },
+      onEnter: onEnter && { events: toEventJson(onEnter) },
+    };
+  }
 }
 
-function getEventJson(events: Event[]): CfnDetectorModel.EventProperty[] {
+function toEventJson(events: Event[]): CfnDetectorModel.EventProperty[] {
   return events.map(e => {
     return {
       eventName: e.eventName,
       condition: e.condition?.evaluate(),
     };
   });
 }
+
+function toTransitionEventJson(events: TransitionEvent[]): CfnDetectorModel.TransitionEventProperty[] | undefined {
```

Comment:
Since this takes an array, it should probably be called "toTransitionEvent**s**Json" instead.